vue-tsc 3.3.6: `UButton`'s click event only allows void return types

### Environment

Nuxt project info:
|                      |                                                               |
| -------------------- | ------------------------------------------------------------- |
| **Operating system** | `Linux 6.1.43`                                                |
| **CPU**              | `AMD EPYC (2 cores)`                                          |
| **Node.js version**  | `v22.20.0`                                                    |
| **nuxt/cli version** | `3.36.1`                                                      |
| **Package manager**  | `pnpm@10.18.3`                                                |
| **Nuxt version**     | `4.4.8`                                                       |
| **Nitro version**    | `2.13.4`                                                      |
| **Builder**          | `vite@7.3.6`                                                  |
| **Config**           | `compatibilityDate`, `css`, `devtools`, `future`, `modules`   |
| **Modules**          | `@nuxt/ui@4.9.0`                                              |

### Is this bug related to Nuxt or Vue?

Nuxt

### Package

v4.x

### Version

v4.9.0

### Reproduction

https://codesandbox.io/p/devbox/nostalgic-currying-dp8xxz

The template linked in the issue creation workflow is pretty outdated and broken by the way, it took some time to get it to work. Could this maybe get updated so it's easier to create repros? 🙏🏻

### Description

Hello,

since updating vue-tsc to the latest version (v3.3.6) click event handlers passed to `@click` of `UButton` only allow void return types. This prevents the usage of simple inline logic like `@click="counter = counter + 1"`. To fix the type error the handler now needs to be wrapped in `void` like `@click="void (counter = counter + 1)"`.

I am unsure if this is a bug in vue-tsc or a valid warning since the return type is typed too strict.

It is probably caused by this change to vue-tsc: https://github.com/vuejs/language-tools/commit/9b6fb7fc787f8c5fafa7a315e79431eb8d466a45

```
app/pages/index.vue:8:15 - error TS2322: Type '(event: MouseEvent) => number' is not assignable to type '((event: MouseEvent) => void | Promise<void>) | ((event: MouseEvent) => void | Promise<void>)[] | undefined'.
  Type '(event: MouseEvent) => number' is not assignable to type '(event: MouseEvent) => void | Promise<void>'.
    Type 'number' is not assignable to type 'void | Promise<void>'.

    <UButton @click="counter = counter + 1"> CLICK </UButton>
```


### Additional context

_No response_

### Logs

```shell-script

```

--- a/src/runtime/components/BlogPost.vue
+++ b/src/runtime/components/BlogPost.vue
@@ -41,7 +41,7 @@ export interface BlogPostProps {
   variant?: BlogPost['variants']['variant']
   to?: LinkProps['to']
   target?: LinkProps['target']
-  onClick?: (event: MouseEvent) => void | Promise<void>
+  onClick?: (event: MouseEvent) => void
   class?: any
   ui?: BlogPost['slots']
 }

--- a/src/runtime/components/Button.vue
+++ b/src/runtime/components/Button.vue
@@ -31,7 +31,7 @@ export interface ButtonProps extends UseComponentIconsProps, Omit<LinkProps, 'ra
   block?: boolean
   /** Set loading state automatically based on the `@click` promise state */
   loadingAuto?: boolean
-  onClick?: ((event: MouseEvent) => void | Promise<void>) | Array<((event: MouseEvent) => void | Promise<void>)>
+  onClick?: ((event: MouseEvent) => void) | Array<((event: MouseEvent) => void)>
   class?: any
   ui?: Button['slots']
 }

--- a/src/runtime/components/ChangelogVersion.vue
+++ b/src/runtime/components/ChangelogVersion.vue
@@ -37,7 +37,7 @@ export interface ChangelogVersionProps {
   indicator?: boolean
   to?: LinkProps['to']
   target?: LinkProps['target']
-  onClick?: (event: MouseEvent) => void | Promise<void>
+  onClick?: (event: MouseEvent) => void
   class?: any
   ui?: ChangelogVersion['slots']
 }

--- a/src/runtime/components/Form.vue
+++ b/src/runtime/components/Form.vue
@@ -62,7 +62,7 @@ export type FormProps<S extends FormSchema, T extends boolean = true, N extends 
   loadingAuto?: boolean
   class?: any
   ui?: { base?: any }
-  onSubmit?: ((event: FormSubmitEvent<FormData<S, T>>) => void | Promise<void>) | (() => void | Promise<void>)
+  onSubmit?: ((event: FormSubmitEvent<FormData<S, T>>) => void) | (() => void)
 } & /** @vue-ignore */ Omit<FormHTMLAttributes, 'name'>
 
 export interface FormEmits<S extends FormSchema, T extends boolean = true> {

--- a/src/runtime/components/LinkBase.vue
+++ b/src/runtime/components/LinkBase.vue
@@ -5,7 +5,7 @@ export interface LinkBaseProps {
   as?: string
   type?: string
   disabled?: boolean
-  onClick?: ((e: MouseEvent) => void | Promise<void>) | Array<((e: MouseEvent) => void | Promise<void>)>
+  onClick?: ((e: MouseEvent) => void) | Array<((e: MouseEvent) => void)>
   href?: string | null
   navigate?: (e: MouseEvent) => void
   target?: LinkProps['target']

--- a/src/runtime/components/PageCard.vue
+++ b/src/runtime/components/PageCard.vue
@@ -53,7 +53,7 @@ export interface PageCardProps {
   variant?: PageCard['variants']['variant']
   to?: LinkProps['to']
   target?: LinkProps['target']
-  onClick?: (event: MouseEvent) => void | Promise<void>
+  onClick?: (event: MouseEvent) => void
   class?: any
   ui?: PageCard['slots']
 }

--- a/src/runtime/components/PageFeature.vue
+++ b/src/runtime/components/PageFeature.vue
@@ -28,7 +28,7 @@ export interface PageFeatureProps {
   orientation?: PageFeature['variants']['orientation']
   to?: LinkProps['to']
   target?: LinkProps['target']
-  onClick?: (event: MouseEvent) => void | Promise<void>
+  onClick?: (event: MouseEvent) => void
   class?: any
   ui?: PageFeature['slots']
 }

--- a/src/runtime/components/User.vue
+++ b/src/runtime/components/User.vue
@@ -30,7 +30,7 @@ export interface UserProps {
   orientation?: User['variants']['orientation']
   to?: LinkProps['to']
   target?: LinkProps['target']
-  onClick?: (event: MouseEvent) => void | Promise<void>
+  onClick?: (event: MouseEvent) => void
   class?: any
   ui?: User['slots']
 }

--- a/src/runtime/vue/overrides/inertia/LinkBase.vue
+++ b/src/runtime/vue/overrides/inertia/LinkBase.vue
@@ -5,7 +5,7 @@ export interface LinkBaseProps {
   as?: string
   type?: string
   disabled?: boolean
-  onClick?: ((e: MouseEvent) => void | Promise<void>) | Array<((e: MouseEvent) => void | Promise<void>)>
+  onClick?: ((e: MouseEvent) => void) | Array<((e: MouseEvent) => void)>
   href?: string
   target?: LinkProps['target']
   rel?: LinkProps['rel']

